sdk/core/azure-core/src/main/java/com/azure/core/util/AttributesBuilder.java

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util;
+
+/**
+ * Generic attribute collection applicable to metrics, tracing and logging implementations.
+ * Implementation is capable of handling different attribute types, caching and optimizing the internal representation.
+ */
+public interface AttributesBuilder {
```

Comment:
We actually lack attribute abstraction for traces and logs. In case of otel, the underlying implementation is the same across all signals and I added one here to make sure we'll be able to share them later. 

Dimensions are exactly the same as attributes, but very specific to metrics.
I agreee that `Attribute` is too wide term to live in `util`. Perhaps `TelemetryAttributes` or `ObservabilityAttributes` would work better?

Reply:
Both sounds good. `TelemetryAttributes` is easier to say (at least for me).